### PR TITLE
DocDown Tests

### PR DESCRIPTION
**Pull request by danielfarrell**
Split out a new file reducer to handle the actions on an individual
file and add a test for it.  Some refactoring to allow for easier
testing. Still need to remove side effects from the reducers to allow
the file testing in there.

### CODE REVIEW

File: app/containers/FileTree.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ function buildTree(folderName, files, handleClick, forceOpen = false) {
         return <File key={value} name={filename} path={value} handleClick={handleClick} />;
       }
       return <UnsupportedFile key={value} name={filename} />;
-    }).valueSeq();
+    }).valueSeq().toArray();
```
Comment (wpultz) [inline app/containers/FileTree.js]: Is this necessary? I would have thought the iterable produced by `.valueSeq()` would be viable for component children.

Comment (danielfarrell) [inline app/containers/FileTree.js]: It's not "necessary" really, but to make the prop types be an array of elements it is.

Comment (wpultz) [inline app/containers/FileTree.js]: Gotcha; without using react-immutable-proptypes, writing a PropTypes for an Immutable.List of elements would be un-fun.

File: app/reducers/files.js [hunk 2]
```diff
@@ -118,99 +94,60 @@ const filesReducer = handleActions({
...
-    let file = state.getIn(['files', id], Map({}));
-    file = file.merge({
-      contents: text || '',
-      changed: (file.get('changed') || file.get('contents') !== text)
-    });
+  [FILE_UPDATE]: (state, action) => {
+    const id = state.get('currentFile');
+    let file = state.getIn(['files', id]);
+    file = fileReducer(file, action);
     return state.setIn(['files', id], file);
```
Comment (wpultz) [inline app/reducers/files.js]: I see a couple instances of pulling something out of state, changing it, then setting it back into the state. I might be more concise to make use of `.update`. Something like... ```return state.updateIn(['files', id], file => fileReducer(file, action))```

Comment (danielfarrell) [inline app/reducers/files.js]: Oh nice, I did not know about update/updateIn.

**Pull request merged**